### PR TITLE
MWPW-125274 - UAR Analytics

### PR DESCRIPTION
**Pull request by sabyamon**
- Adding analytics data for the next button.
- Updated the URL on card click.
- Refactored some code to use better naming conventions.
- Passing the analytics data to the result page as part of local storage.

image

image

Resolves: https://jira.corp.adobe.com/browse/MWPW-125274

**Test URLs:**
- Before: https://uar-integration--milo--adobecom.hlx.page/drafts/sabya/quiz-2/?martech=off
- After: https://uar-analytics--milo--sabyamon.hlx.page/drafts/sabya/quiz-2/?martech=off

### CODE REVIEW

**Comment by echampio-at-adobe** [comment]
I wouldn't suggest such changes now, but afterwards, for the work of preserving state in the URL, it seems to me as if your current urlParam variable is your real application state. I think the useReducer hook might be helpful in the long run. https://react.dev/reference/react/useReducer

File: libs/blocks/quiz/quiz.js [hunk 1]
```diff
@@ -57,45 +60,100 @@ const App = () => {
       const currentflow = userFlow.shift();
       if (!currentflow.length) {
         console.log('No next view so setting select question to empty');
```
Comment (honstar) [inline libs/blocks/quiz/quiz.js:62]: Just noticing `console.log` which should not be here.

**Comment by honstar** [comment]
Overall, I see a lot of new code being added but no additional tests that cover it. Can you please check that code coverage is taken care of @sabyamon ? Thx

File: libs/blocks/quiz/quiz.js [hunk 2]
```diff
@@ -57,45 +60,100 @@ const App = () => {
       const currentflow = userFlow.shift();
```
Comment (JackySun9) [inline libs/blocks/quiz/quiz.js:60]: should be currentFlow and should we have comments for this function too?

File: libs/blocks/quiz/quiz.js [hunk 3]
```diff
@@ -22,6 +22,9 @@ const App = () => {
   const [currentStep, setCurrentStep] = useState(0);
   const [totalSteps, setTotalSteps] = useState(3);
   const [prevStepIndicator, setPrevStepIndicator] = useState([]);
+  const [nextQuizViewsExist, setNextQuizViewsExist] = useState(true);
+  const [urlParam, setUrlParam] = useState({});
+  const [btnAnalytics, setBtnAnalytics] = useState(null);
 
   useEffect(() => {
     (async () => {
```
Comment (JackySun9) [inline libs/blocks/quiz/quiz.js:30]: should be dataStrings and should we have comments for this function too?

**Review by JackySun9 — CHANGES_REQUESTED** [review]

File: libs/blocks/quiz/quiz.js (hunk 2 above)
Comment (sabyamon) [reply libs/blocks/quiz/quiz.js:60]: I have not added comments for all the functions knowingly. Just for the ones which might be a little heavy on logic and Wiki knowledge. Will change the variable names.

File: libs/blocks/quiz/quiz.js (hunk 3 above)
Comment (sabyamon) [reply libs/blocks/quiz/quiz.js:30]: I have not added comments for all the functions knowingly. Just for the ones which might be a little heavy on logic and Wiki knowledge. Will change the variable names.

File: libs/blocks/quiz/quiz.js (hunk 1 above)
Comment (sabyamon) [reply libs/blocks/quiz/quiz.js]: Thanks for noticing it. Removed.

**Review by elaineskpt — APPROVED** [review]

File: libs/blocks/quiz/utils.js [hunk 4]
```diff
@@ -15,25 +16,25 @@ const initConfigPath = (roolElm) => {
...
-  getQuizKey = initQuizKey(rootElement);
-  getAnalyticsType = initAnalyticsType();
-  getAnalyticsQuiz = initAnalyticsQuiz();
+  configPath = initConfigPath(rootElement);
+  quizKey = initQuizKey(rootElement);
+  analyticsType = initAnalyticsType();
+  analyticsQuiz = initAnalyticsQuiz();
 };
 
 export const getQuizData = async () => {
```
Comment (JackySun9) [inline libs/blocks/quiz/utils.js:40]: Should be 'dataStrings' and remove console.log and use Lana to log it

Comment (sabyamon) [reply libs/blocks/quiz/utils.js:40]: @JackySun9 - Changed the variable name. There are a lot of console.log instances already in the catch blocks throughout Milo codebase. We have a separate story to dispatch to LANA.

**Review by JackySun9 — APPROVED** [review]

**Pull request merged**